packages/react-code-editor/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -437,7 +437,7 @@ export class CodeEditor extends React.Component<CodeEditorProps, CodeEditorState
                   {editorHeader}
                   <div className={css(styles.codeEditorMain)}>
                     <input {...getInputProps()} /* hidden, necessary for react-dropzone */ />
-                    {showEmptyState && !value ? emptyState : editor}
+                    {showEmptyState || providedEmptyState && !value ? emptyState : editor}
```

Comment:
The `&&` evaluates first. I think you meant `(showEmptyState || providedEmptyState) && !value ? emptyState : editor`?